### PR TITLE
Release scanner if global status is not ok after calling scanner.open

### PR DESCRIPTION
**Pull request by caneGuy**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

File: be/src/exec/vectorized/hdfs_scan_node.cpp [hunk 1]
```diff
@@ -385,6 +382,11 @@ void HdfsScanNode::_scanner_thread(HdfsScanner* scanner) {
 
     Status status = scanner->open(_runtime_state);
     scanner->set_keep_priority(false);
+    // if global status was not ok, we need fast failure and no need to read file
```
Comment (DorianZheng) [inline be/src/exec/vectorized/hdfs_scan_node.cpp:385]: check this in the while-loop of LOC:396

**Comment by caneGuy** [comment]
run starrocks_clang-format

**Comment by caneGuy** [comment]
run starrocks_clang-format

**Review by dirtysalt — DISMISSED** [review]

File: be/src/exec/vectorized/hdfs_scan_node.cpp [hunk 2]
```diff
@@ -392,6 +389,12 @@ void HdfsScanNode::_scanner_thread(HdfsScanner* scanner) {
     ChunkPtr chunk = nullptr;
 
     while (status.ok()) {
+        // if global status was not ok, we need fast failure and no need to read file
+        if (!_get_status().ok()) {
+            _release_scanner(scanner);
```
Comment (dirtysalt) [inline be/src/exec/vectorized/hdfs_scan_node.cpp:394]: Could you refactor it to `break`?  And following code will handle `_get_status().ok()`.  I want to the whole workflow simple as possible.

**Review by dirtysalt — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by caneGuy** [comment]
run starrocks_be_unittest

**Pull request merged**